include/boost/basic_any.hpp

```diff
@@ -0,0 +1,440 @@
+                    left.man = right->man;
+                    break;
+                case AnyCast:
+                    return boost::typeindex::type_id<ValueType>() == *info ?
+                            static_cast<BOOST_DEDUCED_TYPENAME remove_cv<ValueType>::type *>(left.content.large_value) : 0;
+                case UnsafeCast:
+                    return reinterpret_cast<BOOST_DEDUCED_TYPENAME remove_cv<ValueType>::type *>(left.content.large_value);
+                case Typeinfo:
+                    return const_cast<void*>(static_cast<const void*>(&boost::typeindex::type_id<ValueType>().type_info()));
+            }
+
+            return 0;
+        }
+
+
+        template <typename ValueType>
+        struct is_small_object_impl : boost::integral_constant<bool, sizeof(ValueType) <= OptimizeForSize &&
+            boost::alignment_of<ValueType>::value <= OptimizeForAlignment &&
+            boost::is_nothrow_move_constructible<ValueType>::value>
+        {};
+
+        template <typename ValueType>
+        struct is_small_object : is_small_object_impl<ValueType>
+        {};
+
+        template <typename ValueType, typename DecayedType = BOOST_DEDUCED_TYPENAME boost::decay<const ValueType>::type>
+        static void create(basic_any& any, const ValueType& value, boost::true_type)
+        {
+            any.man = &small_manager<DecayedType>;
+            new (&any.content.small_value) ValueType(value);
+        }
+
+        template <typename ValueType, typename DecayedType = BOOST_DEDUCED_TYPENAME boost::decay<const ValueType>::type>
+        static void create(basic_any& any, const ValueType& value, boost::false_type)
+        {
+            any.man = &large_manager<DecayedType>;
+            any.content.large_value = new DecayedType(value);
+        }
+
+#ifndef BOOST_NO_CXX11_RVALUE_REFERENCES
+        template <typename ValueType, typename DecayedType = BOOST_DEDUCED_TYPENAME boost::decay<const ValueType>::type>
+        static void create(basic_any& any, ValueType&& value, boost::true_type)
+        {
+            any.man = &small_manager<DecayedType>;
+            new (&any.content.small_value) DecayedType(static_cast<ValueType&&>(value));
+        }
+
+        template <typename ValueType, typename DecayedType = BOOST_DEDUCED_TYPENAME boost::decay<const ValueType>::type>
+        static void create(basic_any& any, ValueType&& value, boost::false_type)
+        {
+            any.man = &large_manager<DecayedType>;
+            any.content.large_value = new DecayedType(static_cast<ValueType&&>(value));
+        }
+#endif
+        public: // non-type template parameters accessors
+            static BOOST_CONSTEXPR_OR_CONST std::size_t buffer_size = OptimizeForSize;
+            static BOOST_CONSTEXPR_OR_CONST std::size_t buffer_align = OptimizeForAlignment;
+
+        public: // structors
+
+        basic_any() BOOST_NOEXCEPT
```

Comment:
Add constexpr to force static initialization for default constructed any.

Reply:
Done